File: boxes.py
from PySide6.QtWidgets import *
from PySide6.QtCore import *

import sys
class VerticalBox(QWidget):
    def __init__(self):
        super().__init__()
        
        self.box= QGroupBox()
        self.box_layout= QVBoxLayout()
        self.frame = QFrame()
        self.frame_layout = QVBoxLayout()
        self.frame.setLayout(self.frame_layout)
        self.box_layout.addWidget(self.frame)
        self.box.setLayout(self.box_layout)
        
    def add_widgets(self, widgets:list[QWidget])-> None:
        
        for widget in widgets:
            self.frame_layout.addWidget(widget)
    
    def get_box(self)->QGroupBox:
        return self.box
        
class HorizontalBox(QWidget):
    def __init__(self):
        super().__init__()
        
        self.box= QGroupBox()
        self.box_layout= QHBoxLayout()
        self.frame = QFrame()
        self.frame_layout = QHBoxLayout()
        self.frame.setLayout(self.frame_layout)
        self.box_layout.addWidget(self.frame)
        self.box.setLayout(self.box_layout)
        
    def add_widgets(self, widgets:list[QWidget])-> None:
        
        for widget in widgets:
            self.frame_layout.addWidget(widget)
    
    def get_box(self)->QGroupBox:
        return self.box
    

File: boxes/boxes.cpp
#include <boxes.h>
#include <QtCore>
#include <QWidget>
#include <QObject>
#include <QMainWindow>
#include <QDockWidget>
#include <vector>
#include <QString>
#include <QGroupBox>
#include <QVBoxLayout>
#include <QFrame>

//This is the implementation for the Vertical Box
VerticalBox::VerticalBox(QWidget *parent, const QString& title): QWidget{parent}
{
    main = new QVBoxLayout(this);

    box = new QGroupBox(title, this);
    boxLayout = new QVBoxLayout(box);

    frame = new QFrame(box);
    frameLayout = new QVBoxLayout(frame);

    box->setLayout(boxLayout);
    boxLayout->addWidget(frame);

    
    main->addWidget(box);

}

void VerticalBox::addWidgets(const std::vector<QWidget*> &widgets)
{
    for (QWidget *widget : widgets)
    {
        if (widget)
        {
            frameLayout->addWidget(widget);
        }
        
    }

}
